landoapi/api/landings.py

```diff
@@ -118,6 +121,13 @@ def update(landing_id, data):
         revision (sha) of push if landed == true
         empty string if landed == false
     """
+    if request.headers['API-Key'] != TRANSPLANT_API_KEY:
```

Comment:
you should be doing constant time string comparison to protect against password timing attacks.
there also needs to be some form of brute force detection/protection.